### PR TITLE
feat: add runbook for otel-collector gateway

### PR DESCRIPTION
**Pull request by madwort**
* this will enable us to send spans from job-runner via job-server into honeycomb
* use the distributed docker image
* configure https & http basic auth for receiver
* configure honeycomb key for exporter
* fixes https://github.com/opensafely-core/sysadmin/issues/69

### CODE REVIEW

File: docs/otel-collector-runbook.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+exporters:
+  logging:
+    logLevel: info
+
+  otlphttp:
+    endpoint: "https://api.honeycomb.io"
+    headers:
+      "x-honeycomb-team": "${HONEYCOMB_KEY}"
+# required to support metrics spans
```
Comment (bloodearnest) [inline docs/otel-collector-runbook.md:44]: interesting, why is this required for metrics?

**Review by bloodearnest — APPROVED** [review]
Could we also have a copy of config.yaml in the repo?

otel collector can do env var substitution in the yaml config, so we could maybe do sth like:

```
   "x-honeycomb-team": "${HONEYCOMB_API_KEY}"
```

so that we can just use this config?

**Comment by madwort** [comment]
oh yeah, I did add envvar substitution, the first example config.yaml includes that https://github.com/opensafely-core/sysadmin/pull/74/files#diff-db1ff04c86ec664a41dbd7b2151f74d909bd6c5713900c9b13c7c13238bfc525R43 but the second one doesn't as I neglected to duplicate the change. I'll add a copy of the final file to the repo too.

File: docs/otel-collector-runbook.md (hunk 1 above)
Comment (bloodearnest) [reply docs/otel-collector-runbook.md:44]: interesting. I think if this collector was generated the metrics, that's makes, but for metrics generated elsewhere and sent here to be forwarded, we'd want the source collector to set the dataset?
